### PR TITLE
  Validator Set Override Mechanism

### PR DESCRIPTION
**Pull request by lucca30**
Adds configuration-based validator set overrides to BorConfig, allowing specific validators during designated block ranges. Replaces hardcoded validator checks with JSON-configurable OverrideValidatorSetInRange in chain specs. Updates consensus functions (GetSignerSuccessionNumber, Difficulty, SafeDifficulty) to validate override validators. Includes 18 unit tests covering edge cases and the mainnet scenario (blocks 80440819-80440834).

### CODE REVIEW

File: polygon/heimdall/validator_set_override_test.go [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+	t.Logf("Normal validator 1 succession: %d", succession1)
+
+	succession2, err := validatorSet.GetSignerSuccessionNumber(normalValidator2, blockNumber, config)
+	require.NoError(t, err)
+	t.Logf("Normal validator 2 succession: %d", succession2)
+
+	// Override validator is allowed but gets a calculated succession number
+	overrideSuccession, err := validatorSet.GetSignerSuccessionNumber(overrideValidator, blockNumber, config)
+	require.NoError(t, err)
+	t.Logf("Override validator succession: %d (validator set size: %d)", overrideSuccession, len(validatorSet.Validators))
```
Comment (manav2401) [inline polygon/heimdall/validator_set_override_test.go:492]: Small nit: might be worth to assert the succession number in each case. If not all, specifically the last case to confirm that it returns "-1" (I assume).

**Review by manav2401 — APPROVED** [review]

File: polygon/heimdall/validator_set_override_test.go (hunk 1 above)
Comment (lucca30) [reply polygon/heimdall/validator_set_override_test.go:492]: not actually -1 because it do also consider proposer index. But yeah tests are now included
